core/modules/qana/RelationGraph.cc

```diff
@@ -927,31 +923,53 @@ RelationGraph::RelationGraph(QueryContext const& ctx,
         throw QueryNotEvaluableError(
             "Query must include at least one table reference");
     }
-    double overlap = ctx.css->getDbStriping(ctx.dominantDb).overlap;
     // Build a graph for the first entry in the from list
-    RelationGraph g(ctx, refs.front(), overlap, pool);
+    RelationGraph g(refs.front(), pool);
     // "SELECT ... FROM A, B, C, ..." is equivalent to
     // "SELECT ... FROM ((A CROSS JOIN B) CROSS JOIN C) ..."
     typedef TableRefList::const_iterator Iter;
     for (Iter i = ++refs.begin(), e = refs.end(); i != e; ++i) {
-        RelationGraph tmp(ctx, *i, overlap, pool);
-        g._fuse(JoinRef::CROSS, false, JoinSpec::Ptr(), overlap, tmp);
+        RelationGraph tmp(*i, pool);
```

Comment:
I think you could make this a range-based for loop while you're here, if you feel like it.

Reply:
I thought about it but decided to leave it for a larger refactoring later, there are many loops which can be replaced with range `for`.

Reply:
👍